### ISSUE TITLE
Tilde semver misinterpreted?

### ISSUE THREAD

**Issue by mortenpi** [JuliaLang/Pkg.jl#847]
Not sure if this is a documentation or an implementation issue but the [docs for Pkg](https://julialang.github.io/Pkg.jl/latest/compatibility.html#Tilde-specifiers-1) say that `~1.2.3 = [1.2.3, 1.2.4)`, whereas

```
julia> Pkg.Types.semver_spec("~1.2.3")
VersionSpec("1.2.3-1.2")
```

where, as far as I can tell, `VersionSpec("1.2.3-1.2") = [1.2.3, 1.3.0)`.

**Comment by fredrikekre** [comment]
I think this is a doc issue. I am pretty sure we should mirror cargo here so it should allow updates up to the next minor release, see https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#tilde-requirements.

**Comment by fredrikekre** [comment]
It also seem rather useless to specify that "I allow updating to the next bugfix release, but not the one after that" so I think the implemented behavior makes sense.

**Comment by fredrikekre** [comment]
Not sure why GitHub did not autoclosed this, but fixed by #850
